diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8d5d807..48aea2e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -35,6 +35,11 @@ set_target_properties(appLocalTranslate PROPERTIES
 )
 
 target_compile_options(appLocalTranslate PRIVATE ${MECAB_CFLAGS})
+# This regex checks if the processor is "aarch64" OR starts with "arm"
+if(CMAKE_SYSTEM_PROCESSOR MATCHES "^(aarch64|arm)")
+    message(STATUS "ARM architecture detected (${CMAKE_SYSTEM_PROCESSOR}), adding ARM definition.")
+    target_compile_definitions(appLocalTranslate PRIVATE ARM)
+endif()
 
 target_link_libraries(appLocalTranslate
     PRIVATE Qt6::Quick kotki marian-lite yaml-cpp ICU::uc ICU::i18n ${MECAB_LIBS})
