diff --git a/cmake/CompilerStuff.cmake b/cmake/CompilerStuff.cmake
index 1402994c..642abf75 100644
--- a/cmake/CompilerStuff.cmake
+++ b/cmake/CompilerStuff.cmake
@@ -48,7 +48,12 @@ if(BUILD_ARCH STREQUAL "native")
         list(APPEND INTRINSICS_NVCC -Xcompiler\ -mavx512f)
     endif(AVX512_FOUND)
 else()
-    set(INTRINSICS "-msse4.1")
+    string(TOLOWER "${CMAKE_SYSTEM_PROCESSOR}" CMAKE_SYSTEM_PROCESSOR_LOWER)
+    if(CMAKE_SYSTEM_PROCESSOR_LOWER MATCHES "(aarch64|arm64)")
+        set(INTRINSICS "")
+    else()
+        set(INTRINSICS "-msse4.1")
+    endif()
 endif()
 
 if (CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 9.0)
